File: DSA/LakshaKundra_590013237/Day_79_Question_1.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def inorder(root):
    if root:
        inorder(root.left)
        print(root.val, end=" ")
        inorder(root.right)

root = Node(5)
root.left = Node(3)
root.right = Node(7)
root.left.left = Node(2)
root.left.right = Node(4)
inorder(root)
print()

root2 = Node(10)
root2.left = Node(8)
root2.right = Node(12)
inorder(root2)
print()

File: DSA/PriyankaSingh_590016827/day79question1.c
#include <stdio.h>
#include <stdlib.h>

// Definition for a binary tree node.
typedef struct TreeNode {
    int data;
    struct TreeNode* left;
    struct TreeNode* right;
} TreeNode;

// Function to create a new tree node
TreeNode* createNode(int data) {
    TreeNode* newNode = (TreeNode*)malloc(sizeof(TreeNode));
    newNode->data = data;
    newNode->left = NULL;
    newNode->right = NULL;
    return newNode;
}

// Minimum in BST
int findMin(TreeNode* root) {
    if (root == NULL) {
        return -1; // Or some other indicator for an empty tree
    }
    TreeNode* current = root;
    while (current->left != NULL) {
        current = current->left;
    }
    return current->data;
}

// Maximum in BST
int findMax(TreeNode* root) {
    if (root == NULL) {
        return -1; // Or some other indicator for an empty tree
    }
    TreeNode* current = root;
    while (current->right != NULL) {
        current = current->right;
    }
    return current->data;
}

int main() {
    // Example 1
    TreeNode* root1 = createNode(8);
    root1->left = createNode(3);
    root1->right = createNode(10);
    root1->left->left = createNode(1);
    root1->left->right = createNode(6);
    root1->right->right = createNode(14);
    printf("Min = %d, Max = %d\n", findMin(root1), findMax(root1));

    // Example 2
    TreeNode* root2 = createNode(5);
    printf("Min = %d, Max = %d\n", findMin(root2), findMax(root2));

    return 0;
}
